packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -0,0 +1,373 @@
+        manifestData,
+        this.bucket,
+      );
+
+      const jobEntity = await this.jobRepository.create({
+        chainId,
+        userId,
+        manifestUrl,
+        manifestHash,
+        status: JobStatus.PENDING,
+        waitUntil: new Date(),
+      });
+
+      if (!jobEntity) {
+        this.logger.log(ErrorJob.NotCreated, JobService.name);
+        throw new NotFoundException(ErrorJob.NotCreated);
+      }
+
+      await this.paymentService.savePayment(
+        userId,
+        PaymentSource.BALANCE,
+        PaymentType.WITHDRAWAL,
+        BigNumber.from(totalAmount),
+      );
+
+      jobEntity.status = JobStatus.PAID;
+      await jobEntity.save();
+
+      return jobEntity.id;
+  }
+
+  public async createCvatJob(userId: number, dto: JobCvatDto): Promise<number> {
+    const {
+      chainId,
+      dataUrl,
+      annotationsPerImage,
+      labels,
+      requesterDescription,
+      requesterAccuracyTarget,
+      fundAmount,
+    } = dto;
+
+    const userBalance = await this.paymentService.getUserBalance(userId);
+
+    const fundAmountInWei = ethers.utils.parseUnits(
+      fundAmount.toString(),
+      'ether',
+    );
+
+    const totalFeePercentage = BigNumber.from(this.configService.get<number>(
+      'JOB_LAUNCHER_FEE',
+      0,
+    ))
+      .add(this.configService.get<number>(
+        'RECORDING_ORACLE_FEE',
+        0,
+      ))
+      .add(this.configService.get<number>(
+        'REPUTATION_ORACLE_FEE',
+        0,
+      ));
+    const totalFee = BigNumber.from(fundAmountInWei)
+      .mul(totalFeePercentage)
+      .div(100);
+    const totalAmount = BigNumber.from(fundAmountInWei).add(totalFee);
+
+    if (userBalance.lte(totalAmount)) {
+      this.logger.log(ErrorJob.NotEnoughFunds, JobService.name);
+      throw new NotFoundException(ErrorJob.NotEnoughFunds);
+    }
+
+    const manifestData: ManifestDto = {
+      dataUrl,
+      submissionsRequired: annotationsPerImage,
+      labels,
+      requesterDescription,
+      requesterAccuracyTarget,
+      fundAmount: totalAmount.toString(),
+      mode: JobMode.BATCH,
+      requestType: JobRequestType.IMAGE_LABEL_BINARY,
+    };
+
+    const { manifestUrl, manifestHash } = await this.saveManifest(
+      manifestData,
+      this.bucket,
+    );
+
+    const jobEntity = await this.jobRepository.create({
+      chainId,
+      userId,
+      manifestUrl,
+      manifestHash,
+      status: JobStatus.PENDING,
+      waitUntil: new Date(),
+    });
+
+    if (!jobEntity) {
+      this.logger.log(ErrorJob.NotCreated, JobService.name);
+      throw new NotFoundException(ErrorJob.NotCreated);
+    }
+
+    await this.paymentService.savePayment(
+      userId,
+      PaymentSource.BALANCE,
+      PaymentType.WITHDRAWAL,
+      BigNumber.from(fundAmount),
+    );
+
+    jobEntity.status = JobStatus.PAID;
+    await jobEntity.save();
+
+    return jobEntity.id;
+  }
+
+  public async launchJob(jobEntity: JobEntity): Promise<JobEntity> {
```

Comment:
what if manifest url is invalid? We launch an escow without manifest. We should check it first.